Modules/WeakCache/WeakCache.swift

```diff
@@ -0,0 +1,59 @@
+// Copyright © 2020 Jamit Labs GmbH. All rights reserved.
+
+import Foundation
+
+///
+/// `WeakCache` is a DataStructure based on `NSHashTable`. The reference counter of the stored elements is not
+/// increased by adding it to the `WeakCache`.
+/// If the element is deinitialized it is automatically removed from the cache.
+///
+/// Example:
+///
+/// ```swift
+/// let cache: WeakCache<UIViewController> = .init()
+///
+/// cache.append(someViewController)
+/// ```
+///
+public final class WeakCache<Element> {
+    private let elements: NSHashTable<AnyObject>
+
+    /// All elements of the cache
+    public var all: [Element] {
+        return elements.allObjects.compactMap { $0 as? Element }
+    }
+
+    public init() {
+        elements = NSHashTable<AnyObject>.weakObjects()
+    }
+
+    /// Appends a new element to the cache
+    ///
+    /// - parameters:
+    ///     - element: The element to append
+    public func append(_ element: Element) {
+        elements.add(element as AnyObject)
+    }
+
+    /// Removes an element from the cache
+    ///
+    /// - parameters:
+    ///     - element: The element to remove
+    public func remove(_ element: Element) {
```

Comment:
I suggest to keep it as is in the first moment since it is possible to use the `all` property. Forwarding functions of the `all` Collection or the `NSHashTable` itself is actual superfluous in my opinion. What do you think?